### PR TITLE
Create AzureLogsCollector module

### LINKED ISSUE #517: Create AzureLogsCollector module
Should save logs in JSON-L format.

### PR DESCRIPTION
**Pull request by jonathan-greig**
Fixes #517

### CODE REVIEW

File: tests/lib/collectors/azure_logging.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+class AzureLogging(unittest.TestCase):
+  """Tests for the Azure logging collector."""
+
+  def testInitialization(self):
+    """Tests that the collector can be initialized."""
+    test_state = state.DFTimewolfState(config.Config)
+    azure_logging_collector = azure_logging.AzureLogsCollector(test_state)
+    self.assertIsNotNone(azure_logging_collector)
+
```
Comment (ramo-j) [inline tests/lib/collectors/azure_logging.py:23]: I know it's tedious, but we should still test `SetUp` in here to make sure attributes are set correctly.

File: tests/lib/collectors/azure_logging.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
```
Comment (ramo-j) [inline tests/lib/collectors/azure_logging.py:1]: I'm a big fan of testing error cases, if it's possible to add some in.

File: dftimewolf/lib/collectors/azure_logging.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+    credentials = az_identity.DefaultAzureCredential()
+    monitoring_client = az_monitor.MonitorManagementClient(
+        credentials, self._subscription_id)
+    activity_logs_client = monitoring_client.activity_logs
+
+    try:
+      results = activity_logs_client.list(filter=self._filter_expression)
+
+      while True:
```
Comment (ramo-j) [inline dftimewolf/lib/collectors/azure_logging.py:57]: What guarantee do we have this won't infinite loop? `StopIteration` *MUST* be thrown at some point?

**Review by ramo-j — CHANGES_REQUESTED** [review]
Just some test cases I'd like to see added mainly, otherwise looks solid to me.

File: dftimewolf/lib/collectors/azure_logging.py (hunk 3 above)
Comment (jonathan-greig) [reply dftimewolf/lib/collectors/azure_logging.py:57]: `StopIteration` will be raised when there are no more log entries to be returned for the query (max time range is 90 days). Putting a limit on the loop would limit the number of log entries that could be collected.

File: tests/lib/collectors/azure_logging.py (hunk 2 above)
Comment (jonathan-greig) [reply tests/lib/collectors/azure_logging.py:1]: Done

**Review by ramo-j — APPROVED** [review]

**Pull request merged**